app/controllers/cases/attachments_controller.rb

```diff
@@ -27,6 +29,34 @@ def destroy
       end
     end
 
+    def new
+      authorize @case, :can_upload_request_attachment?
+      @s3_direct_post = S3Uploader.for(@case, 'requests')
+    end 
+
+    def create 
+      authorize @case, :can_upload_request_attachment?
+      service = RequestUploaderService.new(
+        kase: @case,
+        current_user: current_user,
+        uploaded_files: create_params[:uploaded_request_files],
+        upload_comment: create_params[:upload_comment]
+      )
+      service.upload!
+
+      if service.result == :error
+        @s3_direct_post = S3Uploader.for(@case, 'requests')
+        @case = @case.decorate
+        flash.now[:alert] = service.error_message
+        render :new
+      else
+        if service.result == :ok
+          flash[:notice] = t('notices.request_uploaded')
+        end
+        redirect_to case_path @case
+      end
+    end 
```

Comment:
Wondering if the inner conditional on line 53 ought to be on the same level as the outer conditional from line 47? 

Or is it like this intentionally to handle the `:blank` case from the service? 🤔 

Just looks a little unusual - I think normally for these we do 

```
    if service.result == :ok
      happy things
    else
      render or redirect with error
    end
```

Reply:
I changed it to case statement with extra part for handling ':blank' value